crates/noirc_evaluator/src/brillig/mod.rs

```diff
@@ -46,15 +49,54 @@ impl std::ops::Index<FunctionId> for Brillig {
 }
 
 impl Ssa {
-    /// Generate compilation artifacts for brillig functions
+    /// Compile to brillig brillig functions and ACIR functions reachable from them
     pub(crate) fn to_brillig(&self) -> Brillig {
-        // Collect all of the brillig functions
-        let brillig_functions =
-            self.functions.values().filter(|func| func.runtime() == RuntimeType::Brillig);
+        // Collect all the function ids that are reachable from brillig
+        // That means all the functions marked as brillig and ACIR functions called by them
+        let mut reachable_function_ids: HashSet<FunctionId> = HashSet::new();
+
+        // Initialize the queue with all the functions marked as brillig
+        let mut reachability_queue: Vec<FunctionId> = self
+            .functions
+            .iter()
+            .filter_map(
+                |(id, func)| {
+                    if func.runtime() == RuntimeType::Brillig {
+                        Some(*id)
+                    } else {
+                        None
+                    }
+                },
+            )
+            .collect();
+
+        while !reachability_queue.is_empty() {
```

Comment:
nit: we can use `while let Some() = reachability_queue.pop() {}` which would then avoid the need to panic, since the panic is only needed because we do do ` while !reachability_queue.is_empty()`

Reply:
nice